public/app/plugins/datasource/grafana-pyroscope-datasource/QueryEditor/QueryEditor.tsx

```diff
@@ -119,29 +126,74 @@ function useLabels(
     from: Math.floor((range?.from.valueOf() || 0) / 5000) * 5000,
   };
 
-  const labelsResult = useAsync(() => {
-    return datasource.getLabelNames(query.profileTypeId + query.labelSelector, unpreciseRange.from, unpreciseRange.to);
-  }, [datasource, query.profileTypeId, query.labelSelector, unpreciseRange.to, unpreciseRange.from]);
+  // Transforms user input into a valid label selector including the profile type.
+  // It can optionally remove a label, used to support editing existing label values.
+  const createSelector = (rawInput: string, profileTypeId: string, labelToRemove: string): string => {
+    let labels: string[] = [`__profile_type__=\"${profileTypeId}\"`];
+    let match;
+    while ((match = labelSelectorRegex.exec(rawInput)) !== null) {
+      if (match[1] && match[2]) {
+        if (match[1] === labelToRemove) {
+          continue;
+        }
+        labels.push(`${match[1]}=${match[2]}`);
+      }
+    }
+    return `{${labels.join(',')}}`;
+  };
+
+  const [availableLabels, setAvailableLabels] = useState(() => ({ data: [''] }));
```

Comment:
Why an object with a `data` property? Why not just the array (some doubt for the lines below)?